lldb/test/API/lang/swift/regex/TestSwiftRegex.py

```diff
@@ -0,0 +1,45 @@
+# TestSwiftRegex.py
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ------------------------------------------------------------------------------
+"""
+Test Swift's regex support
+"""
+import lldb
+from lldbsuite.test.lldbtest import *
+from lldbsuite.test.decorators import *
+import lldbsuite.test.lldbutil as lldbutil
+
+
+class TestSwiftRegex(TestBase):
+    mydir = TestBase.compute_mydir(__file__)
+
+    def setUp(self):
+        TestBase.setUp(self)
+        self.main_source = "main.swift"
+        self.main_source_spec = lldb.SBFileSpec(self.main_source)
+
+    @swiftTest
+    def test_swift_regex(self):
+        """Test Swift's regex support"""
+        self.build()
+        lldbutil.run_to_source_breakpoint(
+            self, 'Set breakpoint here', self.main_source_spec)
+        self.expect('v regex',
+                    substrs=['_StringProcessing.Regex<(Substring, Substring, Substring, Substring)>) regex = {'])
+        self.expect('po regex',
+                    substrs=['Regex<(Substring, Substring, Substring, Substring)>'])
+
+        self.expect('v dslRegex',
+                    substrs=['(_StringProcessing.Regex<Substring>) dslRegex = {'])
+        self.expect('po dslRegex',
+                    substrs=['Regex<Substring>'])
```

Comment:
are these validation tests that check only that `v` and `po` do something? In other words, do we need the `substrs` (which themselves don't seem useful), or can the tests be simplified to `self.expect('v regex')` etc?

Reply:
We always want to check the output. Otherwise the expression might succeed with an empty result and we never catch it.

Reply:
good point. However I do think that in the case of the printing commands (`p`/`v`), an empty output should raise.